### PR TITLE
fix(nuxt3): handle prematurely ended `res` object

### PR DESCRIPTION
**Pull request by danielroe**
### 🔗 Linked issue

related: #4244

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation or readme)
- [x] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

It's possible before these points for the request to have been ended (such as in a previous `app:rendered` or in a router middleware).

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: packages/nuxt3/src/app/composables/cookie.ts [hunk 1]
```diff
@@ -65,7 +65,7 @@ function writeClientCookie (name: string, value: any, opts: CookieSerializeOptio
 }
 
 function writeServerCookie (event: CompatibilityEvent, name: string, value: any, opts: CookieSerializeOptions = {}) {
-  if (event) {
+  if (event && !event.res.writableEnded) {
```
Comment (pi0) [inline packages/nuxt3/src/app/composables/cookie.ts:68]: This is not fixing any issue but silently ignoring a cookie addition that is expected but is not possible anymore!

Comment (danielroe) [reply packages/nuxt3/src/app/composables/cookie.ts:68]: it's an internal function so I think we could easily rename. I would not expect this to error; I think a warning would be okay.

Comment (pi0) [reply packages/nuxt3/src/app/composables/cookie.ts:68]: It is used by `useCookie`. Silent errors are not a fix for lifecycle issues!

**Comment by pi0** [comment]
DX to be improved in h3.

**Pull request closed without merging**